xbmc/cores/AudioEngine/Sinks/pipewire/PipewireGlobal.h

```diff
@@ -29,21 +29,21 @@ class CPipewireGlobal
   CPipewireGlobal() = default;
   ~CPipewireGlobal();
 
-  CPipewireGlobal& SetName(std::string_view name)
+  CPipewireGlobal& SetName(const std::string& name)
   {
     m_name = name;
     return *this;
   }
 
-  std::string_view GetName() const { return m_name; }
```

Comment:
Returning std::string_view is still valid

Reply:
@lrusak  do you agree or am I technically wrong / you have some other strategy in mind?

Reply:
We hold onto a std::string, so returning std::string_view from that should be valid no?

https://en.cppreference.com/w/cpp/string/basic_string/operator_basic_string_view

Reply:
Yes, but why? We have no use of the string view? Most efficient, if inter dependency of lifetime is no issue, is a reference to the string?

Reply:
I think @fritsch is correct. I see it similar to:

```c++
class foo
{
  int* bar() const { return &m_bar; }
  int m_bar;
}; 
```

 Then you'd have UB if you did:
 
 ```c++
foo* x = new foo();
int* y = x.bar();
delete x;
std::cout << *y;
```

Reply:
Does the std::string held by the class not outlive the lifetime of the call?

Reply:
In the code as it is now: Yes, cause it is a Getter that makes a copy, given that the object does not die while the call happens ;-)
In your string_view no. It will invalidate whenever the owning class, which is not the Caller, goes out of scope.

Therefore I suggest merging the code as it is now.

Reply:
So returning a `const std::string&` is also invalid in your mind?

Reply:
One can only return the reference to a value, if the object giving out this reference is 100% guaranteed to outlive the guy holding this reference.

That's why I said: if we know that PipewireGlobal never vanishes before its users, you can return a reference. But see garbear's example, same issue else.

Reply:
And to complete that: the original issue was exactly the same. The object returned a string view referencing an internal local, which died while external users of this object still used it.

Therefore I disconnected the users of the global by simply returning a copy of object's internal member.

Given that PipewireGlobal itself does not vanish while the getter is called, all fine now.

Reply:
Anything still open? That lacks explanation?

Reply:
All fine